pkg/v1/tkg/tkgconfigbom/bom.go

```diff
@@ -451,18 +456,23 @@ var errorDownloadingTKGCompatibilityFile = `failed to download the TKG Compatibi
 If this is an internet-restricted environment please refer to the documentation to set TKG_CUSTOM_IMAGE_REPOSITORY and related configuration variables in %s 
```

Comment:
I think the air-gap/internet-restricted scenario would be broken unless the compatibilityFileLocation provided actually accounted for the repo change that would have been provided via TKG_CUSTOM_IMAGE_REPOSITORY.

If the expectation is that it is the case, the means during the initialization of this compatibilityFileLocation (when no edition exists) or when it is set on an edition switch, TKG_CUSTOM_IMAGE_REPOSITORY will have to be accounted for.

If the expectation is that one can override the config with the 'right' value as a workaround, this error message needs to convey said workaround in the situation where the compatibilityFileLocation was read and used.

Another option is to continue to leverage the TKG_CUSTOM_IMAGE_REPOSITORY handling like how it is done in the else clause in L466 to patch the image path to use the right repository.

Reply:
> Another option is to continue to uniformly leverage the TKG_CUSTOM_IMAGE_REPOSITORY handling like how it is done in the else clause in L466 to patch the compatibilityFileLocation image path to use the right repository.

I'm open to this! Is this the approach that makes the most sense to framework?

If so, we could update the client config to be something like:

```yaml
baseRepoUrl: 
compatibilityPath:
```

Reply:
> > Another option is to continue to uniformly leverage the TKG_CUSTOM_IMAGE_REPOSITORY handling like how it is done in the else clause in L466 to patch the compatibilityFileLocation image path to use the right repository.
> 
> I'm open to this! Is this the approach that makes the most sense to framework?
For the purpose of making options/settings the user sets has consistent behavior, I'd say yes.
e.g. In the past situations where setting TKG_CUSTOM_IMAGE_REPOSITORY should have the desired effect but didn't were treated as bugs and fixed.

> 
> If so, we could update the client config to be something like:
> 
> ```yaml
> baseRepoUrl: 
> compatibilityPath:
> ```
I assume in this proposal, baseRepoURL is something like
projects.registry.vmware.com/tce
and compatibilityPath something like
tkg-compatibility
?

I was originally think just that the current state of TKG_CUSTOM_IMAGE_REPOSITORY be taken into account after reading the compatibility location cli option, but I like the idea of starting to use the config file option to record the equivalent as well. 
If we do the latter, however, we would need to do some refactoring to ensure all places (about 6-7 places) where TKG_CUSTOM_IMAGE_REPOSITORY is read is converted to prefer the client option value if it exists.
Otherwise there will be surprises when values of TKG_CUSTOM_IMAGE_REPOSITORY and the cli option differ for whatever reason.

Reply:
I agree. I think this also makes things more explicit.

I'll try to make this change and evaluate how tough the plumbing is.

Reply:
I like the idea of breaking the `baseRepoUrl` and `compatibilityPath`.